bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaWaterHeatingSystemHandler.java

```diff
@@ -0,0 +1,225 @@
+        stateNames.put(WATER_HEATER_MODE, WATER_HEATER_MODE_STATE);
+
+        stateNames.put(BOOST_MODE_DURATION, BOOST_MODE_DURATION_STATE);
+        // override state type because the cloud sends consumption in percent
+        cacheStateType(BOOST_MODE_DURATION_STATE, TYPE_DECIMAL);
+
+        stateNames.put(AWAY_MODE_DURATION, AWAY_MODE_DURATION_STATE);
+        // override state type because the cloud sends consumption in percent
+        cacheStateType(AWAY_MODE_DURATION_STATE, TYPE_DECIMAL);
+
+        stateNames.put(HEAT_PUMP_OPERATING_TIME, HEAT_PUMP_OPERATING_TIME_STATE);
+        // override state type because the cloud sends consumption in percent
+        cacheStateType(HEAT_PUMP_OPERATING_TIME_STATE, TYPE_DECIMAL);
+
+        stateNames.put(ELECTRIC_BOOSTER_OPERATING_TIME, ELECTRIC_BOOSTER_OPERATING_TIME_STATE);
+        // override state type because the cloud sends consumption in percent
+        cacheStateType(ELECTRIC_BOOSTER_OPERATING_TIME_STATE, TYPE_DECIMAL);
+
+        stateNames.put(POWER_HEAT_PUMP, POWER_HEAT_PUMP_STATE);
+        // override state type because the cloud sends consumption in percent
+        cacheStateType(POWER_HEAT_PUMP_STATE, TYPE_DECIMAL);
+
+        stateNames.put(POWER_HEAT_ELEC, POWER_HEAT_ELEC_STATE);
+        // override state type because the cloud sends consumption in percent
+        cacheStateType(POWER_HEAT_ELEC_STATE, TYPE_DECIMAL);
+    }
+
+    @Override
+    public void updateThingChannels(SomfyTahomaState state) {
+        if (OPERATING_MODE_STATE.equals(state.getName()) && state.getValue() instanceof Map) {
+            logger.debug("Operating Mode State: {}  {}", state.getValue().getClass().getName(), state.getValue());
+
+            Map<String, String> data = (Map<String, String>) state.getValue();
+
+            Object relaunchValue = data.get("relaunch");
+            if (relaunchValue != null) {
+                this.boostMode = relaunchValue.toString().equalsIgnoreCase("on");
+                logger.debug("Boost Value: {}", this.boostMode);
+                updateState(BOOST_MODE, OnOffType.from(this.boostMode));
+            }
+
+            Object awayValue = data.get("absence");
+            if (awayValue != null) {
+                this.awayMode = awayValue.toString().equalsIgnoreCase("on");
+                logger.debug("Away Value: {}", this.awayMode);
+                updateState(AWAY_MODE, OnOffType.from(this.boostMode));
+            }
+        } else if (TARGET_TEMPERATURE_STATE.equals(state.getName())) {
+            logger.debug("Target Temperature: {}", state.getValue());
+            // 50 -> 3
+            // 54.5 -> 4
+            // 62 -> 5
+            Double temp = null;
+            try {
+                temp = Double.parseDouble(state.getValue().toString());
+            } catch (NumberFormatException e) {
+                logger.warn("Unexcepted pre-defined value for Target State Temperature: {}", state.getValue());
+                return;
```

Comment:
By returning, you will not run `super.updateThingChannels(state);`
Th eproper way is probably to put the next lines inside your try